bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -95,99 +145,273 @@ impl Config {
+        unsafe {
+            wasmedge::WasmEdge_ConfigureAddHostRegistration(
+                self.ctx,
+                wasmedge::WasmEdge_HostRegistration_WasmEdge_Process,
+            )
+        };
+        self
+    }
+
+    /// Sets the maximum number of the memory pages available.
+    pub fn set_max_memory_pages(self, count: u32) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureSetMaxMemoryPage(self.ctx, count) };
         self
     }
 
-    /// Get the page limit of memory instances.
+    /// Returns the number of the memory pages available.
     pub fn get_max_memory_pages(&self) -> u32 {
         unsafe { wasmedge::WasmEdge_ConfigureGetMaxMemoryPage(self.ctx) }
     }
 
     // For AOT compiler
 
-    /// Set the optimization level of AOT compiler.
+    /// Sets the optimization level of AOT compiler.
     pub fn set_optimization_level(self, opt_level: CompilerOptimizationLevel) -> Self {
         unsafe {
             wasmedge::WasmEdge_ConfigureCompilerSetOptimizationLevel(self.ctx, opt_level as u32)
         };
         self
     }
 
-    /// Get the optimization level of AOT compiler.
+    /// Returns the optimization level of AOT compiler.
     pub fn get_optimization_level(&self) -> CompilerOptimizationLevel {
         let level = unsafe { wasmedge::WasmEdge_ConfigureCompilerGetOptimizationLevel(self.ctx) };
         level.into()
     }
 
-    /// Set the output binary format of AOT compiler.
+    /// Sets the output binary format of AOT compiler.
     pub fn set_compiler_output_format(self, format: CompilerOutputFormat) -> Self {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerSetOutputFormat(self.ctx, format as u32) };
         self
     }
 
-    /// Get the output binary format of AOT compiler.
+    /// Returns the output binary format of AOT compiler.
     pub fn get_compiler_output_format(&self) -> CompilerOutputFormat {
         let value = unsafe { wasmedge::WasmEdge_ConfigureCompilerGetOutputFormat(self.ctx) };
         value.into()
     }
 
-    /// Set the dump IR boolean value of AOT compiler.
-    pub fn dump_ir(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetDumpIR(self.ctx, enable) };
+    /// Sets the dump IR option of AOT compiler.
+    pub fn dump_ir(self, flag: bool) -> Self {
+        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetDumpIR(self.ctx, flag) };
         self
     }
 
-    /// Get the dump IR option of AOT compiler.
+    /// Checks if the dump IR option turns on or not.
     pub fn is_dump_ir(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerIsDumpIR(self.ctx) }
     }
 
-    /// Set the generic binary option of AOT compiler.
-    pub fn generic_binary(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetGenericBinary(self.ctx, enable) };
+    /// Sets the generic binary option of AOT compiler.
+    pub fn enable_generic_binary(self, flag: bool) -> Self {
```

Comment:
The following `enable_` prefix should also be removed.

Reply:
Accepted, thanks!